src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -18,14 +18,81 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
     return;
   }
 
-  const selectedData: T[] = [];
   const newSelectedRowIds = {} as Record<string, boolean>;
-  instance.initialRows.forEach((row) => {
-    if (newState.selectedRowIds[row.id] && !isRowDisabled?.(row.original)) {
+
+  const handleRow = (row: Row<T>) => {
+    if (isRowDisabled?.(row.original)) {
+      return false;
+    }
+
+    let isAllSubSelected = true;
+    row.initialSubRows.forEach((subRow) => {
+      const result = handleRow(subRow);
```

Comment:
Recursion... 😬

Is react-table using similar recursive logic by default?

Reply:
Yeah, I copied a lot of ideas from there.